controllers/horizon_controller.go

```diff
@@ -547,3 +579,52 @@ func (r *HorizonReconciler) ensureHorizonSecret(
 
 	return nil
 }
+
+func (r *HorizonReconciler) renderMemcached(instance *horizonv1alpha1.Horizon) *memcachedv1.Memcached {
+	return &memcachedv1.Memcached{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "memcached.openstack.org/v1beta1",
+			Kind:       "Memcached",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-memcached", instance.Name),
+			Namespace: instance.Namespace,
+		},
+		Spec: memcachedv1.MemcachedSpec{
+			Replicas: instance.Spec.Replicas,
+		},
+	}
+}
+
+func (r *HorizonReconciler) getMemcachedCR(ctx context.Context, memcachedName string) (bool, error) {
+	list := &memcachedv1.MemcachedList{}
+	if err := r.Client.List(ctx, list); err != nil {
+		r.Log.Info("Error while getting list of memcaches")
+		return false, err
+	}
+	for _, memcached := range list.Items {
+		if memcached.ObjectMeta.Name == memcachedName {
+			r.Log.Info("Found Memcached CR for Horizon!")
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (r *HorizonReconciler) checkMemcachedSvc(memcachedName string, h *helper.Helper) (svc string, err error) {
+	labelSelector := k8s_labels.Set{"app": "memcached"}.String()
+	svcList, err := h.GetKClient().CoreV1().Services("").List(context.TODO(), metav1.ListOptions{
+		LabelSelector: labelSelector,
+	})
+	if err != nil {
+		r.Log.Info("Error getting podlist: %s", err)
+		return "", err
+	}
+	for _, service := range svcList.Items {
+		if service.Name == memcachedName {
+			return service.Name, nil
+		}
+	}
+	r.Log.Info("No Memcached service was found")
+	return "", nil
```

Comment:
as far as I see the usage of checkMemcachedSvc relies on getting the name of the service or getting an error. So you might want to return an error here.

Reply:
Yeah, fair point. Rather than the Info log, it's better to return it as an error. This has been updated in the latest commit.